File: src/app/model/computer.ts
// src/app/models/computer.model.ts

export interface Computer {
    computer_id: string;
    computer_name: string;
    branch_id: string;
    location_in_library?: string;
    computer_type: string;
    specifications?: string;
    operating_system?: string;
    installed_software?: string;
    acquisition_date?: string;
    last_maintenance_date?: string;
    status: string;
    time_limit_minutes: number;
    
    // For display purposes
    branch_name?: string;
  }
  
  export interface ComputerFilter {
    branch_id?: string;
    computer_type?: string;
    status?: string;
    search_term?: string;
  }

File: LibraryManagement.Core/Entities/Computer.cs
using System;
using System.Collections.Generic;

namespace LibraryManagement.Core.Entities;

public partial class Computer
{
    public string ComputerId { get; set; } = null!;

    public string ComputerName { get; set; } = null!;

    public string BranchId { get; set; } = null!;

    public string? LocationInLibrary { get; set; }

    public string ComputerType { get; set; } = null!;

    public string? Specifications { get; set; }

    public string? OperatingSystem { get; set; }

    public string? InstalledSoftware { get; set; }

    public DateOnly? AcquisitionDate { get; set; }

    public DateOnly? LastMaintenanceDate { get; set; }

    public string? Status { get; set; }

    public int? TimeLimitMinutes { get; set; }

    public virtual LibraryBranch Branch { get; set; } = null!;

    public virtual ICollection<ComputerReservation> ComputerReservations { get; set; } = new List<ComputerReservation>();

    public virtual ICollection<ComputerSession> ComputerSessions { get; set; } = new List<ComputerSession>();
}
